exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/registry/FunctionRegistryHolder.java

```diff
@@ -151,6 +151,20 @@ public void removeJar(String jarName) {
     }
   }
 
+  /**
+   * Removes all jars from {@link #jars} map and all associated with jars functions
+   * from {@link #functions}.
+   * This is write operation, so one user at a time can call perform such action,
+   * others will wait till first user completes his action.
+   */
+  public void removeAllJars() {
```

Comment:
To improve performance please consider adding close method that will just close appropriate classloaders and then nullify map holders.

Reply:
Good idea!

I have made this change, could you please review it again?